### PR TITLE
Add foreman proxy scenario

### PR DESCRIPTION
**Pull request by ehelms**
Includes https://github.com/theforeman/foreman-installer/pull/262 with updates

### CODE REVIEW

File: config/foreman-proxy-answers.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# <classname>:
+#   <param>: <value> - include and override the default(s)
+#
+# Every support plugin class is listed, so that it
+# shows up in the interactive menu
+#
+# See params.pp in each class for what options are available
+---
+foreman_proxy:
+  custom_repo: true
```
Comment (ekohl) [inline config/foreman-proxy-answers.yaml:13]: This no longer exists and can be dropped.

**Review by ekohl — CHANGES_REQUESTED** [review]
My issue with this is that we will need to update **all** manuals. They say nothing about passing in a `--scenario` argument. I'd like to hear @tbrisker's opinion about this as well. On the other hand, 2.0 is a good moment for that.

**Comment by tbrisker** [comment]
I'm fine with adding a scenario, tbh I'm a bit surprised that while the installer is able to install just a proxy (and has been for a long while if i'm not mistaken), [the manual](https://theforeman.org/manuals/nightly/index.html#4.3.1SmartProxyInstallation) tells users to manually install it from packages.

**Comment by ekohl** [comment]
That's from the advanced section. [Installation Scenarios](https://theforeman.org/manuals/nightly/index.html#3.2.3InstallationScenarios) describes how to use the installer for it. It basically comes down to passing in `--no-enable-foreman --no-enable-foreman-cli` and you have the same result as this. The benefit for us as maintainers is that we don't need to duplicate all the migrations. We also don't force uses to make an explicit choice whether something is a proxy or a foreman server: there's just components you can enable/disable.

**Comment by ehelms** [comment]
I'm not sure I follow clearly what the argument against this is. As I see it we have two deployments in Foreman:

 1) Foreman Server w/ builtin proxy
 2) External proxy

We should make this obvious and easy for users to install one or the other.

**Comment by ekohl** [comment]
> I'm not sure I follow clearly what the argument against this is

It is additional work for us and by merging this now we break a lot of documentation. First of all, this needs a Redmine issue so it's properly tracked in the release notes. Then we need to make sure our documentation is updated to reflect this before RC1. As long as we can make that happen, I'm good with it.

After that we should also consider how we handle it in forklift. For Katello we also generate `-proxy` boxes. Should we do the same for vanilla and further explode the number of boxes we have?

**Comment by ehelms** [comment]
Ahh yea, I am good with making documentation updates. We technically are not breaking anything by including this. We are adding functionality that we can switch our documentation to as the preferred method for installation of a stand alone proxy but it does not invalidate the current method.

**Comment by ekohl** [comment]
Well, before this change this worked:
```
yum -y install foreman-installer
foreman-installer
```

After it needs to be:
```
yum -y install foreman-installer
foreman-installer --scenario foreman
```

That means documentation changes.

**Pull request closed without merging**